examples/preprocessing/plot_count_featurizer.py

```diff
@@ -0,0 +1,138 @@
+"""
+=========================================================
+Using CountFeaturizer to featurize frequencies
+=========================================================
+
+Shows how to use CountFeaturizer to transform some categorical variables
+into a frequency feature. CountFeaturizer can often be used to reduce
+training time, classification time, and classification error.
+"""
+from __future__ import print_function
+import matplotlib.pyplot as plt
+import numpy as np
+import time
+from sklearn.preprocessing import FunctionTransformer
+from sklearn.preprocessing.data import CountFeaturizer
+from sklearn.preprocessing.data import OneHotEncoder
+from collections import OrderedDict
+from sklearn.datasets import make_classification
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.pipeline import make_pipeline
+from sklearn.pipeline import FeatureUnion
+
+RANDOM_STATE = 30
+
+n_datapoints = 1000
+n_informative = 30
+n_features = 30
+n_redundant = 0
+
+# Generate a binary classification dataset.
+X, y = make_classification(n_samples=n_datapoints, n_features=n_features,
+                           n_clusters_per_class=1, n_informative=n_informative,
+                           n_redundant=n_redundant, random_state=RANDOM_STATE)
+
+# only make these selected features "categorical"
+discretized_features = [0, 1]
+non_discretized_features = \
+    list(set(range(n_features)) - set(discretized_features))
+non_discretized_features_count = \
+    list(set(range(n_features + 1)) - set(discretized_features))
+
+
+def select_non_discrete(X, count=False):
+    """Selects the non-discrete features."""
+    if count:
+        return X[:, non_discretized_features_count]
+    return X[:, non_discretized_features]
+
+
+def select_discrete(X):
+    """Selects the discrete features."""
+    return X[:, discretized_features]
+
+
+def process_discrete(X):
+    """Processes discrete features to make them categorical."""
+    for feature in discretized_features:
+        X_transform_col = (X[:, feature]).astype(int)
+        col_min = min(np.amin(X_transform_col), 0)
+        X[:, feature] = X_transform_col - col_min
+    return X
+
+
+time_start = time.time()
+pipeline_cf = make_pipeline(
```

Comment:
* It is unclear why the pipeline does not contain the classifier. Now the preprocessing steps and the classifier are separated.
Major consequence: Code is hard to understand.
Minor consequence: The time measurements only involve the classification time.

Reply:
One of the premises CountFeaturizer was built on (based on this Microsoft ML article on count featurization https://docs.microsoft.com/en-us/azure/machine-learning/studio-module-reference/data-transformation-learning-with-counts) was that it may be possible to reduce the time training and classification takes compared to something like one hot encoding due to there being less features generated. Here, the reason why the pipeline was split was because it's necessary to separate out the time it took to preprocess the data from the time it takes to train and use the classifier.